client/client.go

```diff
@@ -923,16 +926,28 @@ func (r *NotaryRepository) RotateKey(role string, serverManagesKey bool) error {
 	}
 
 	var (
-		pubKey    data.PublicKey
-		err       error
-		errFmtMsg string
+		pubKey     data.PublicKey
+		pubKeyList data.KeyList
+		err        error
+		errFmtMsg  string
 	)
-	switch serverManagesKey {
-	case true:
+
+	switch {
+	case serverManagesKey:
 		pubKey, err = rotateRemoteKey(r.baseURL, r.gun, role, r.roundTrip)
+		pubKeyList = make(data.KeyList, 0, 1)
+		pubKeyList = append(pubKeyList, pubKey)
 		errFmtMsg = "unable to rotate remote key: %s"
-	default:
+	case !serverManagesKey && len(keyList) > 0:
+		pubKeyList = make(data.KeyList, 0, len(keyList))
+		for _, keyID := range keyList {
+			pubKey = r.CryptoService.GetKey(keyID)
+			pubKeyList = append(pubKeyList, pubKey)
+		}
```

Comment:
@cyli it seems like the `for range` will just abort if the keylist is `nil`

https://play.golang.org/p/QzFKDS5gaa